include/swift/AST/DiagnosticEngine.h

```diff
@@ -1055,6 +1055,8 @@ namespace swift {
     llvm::StringRef diagnosticStringFor(const DiagID id,
                                         bool printDiagnosticNames);
 
+    static llvm::StringRef diagnosticIDStringFor(const DiagID id);
```

Comment:
Could we use this in `LocalizationProducer::getMessageOr` and remove `diagnosticNameStrings` from that file?  Since that function needs to create a copy of the string anyway it seems reasonable to add the surrounding "[]" at that point rather than need two separate tables.  CC @HassanElDesouky @owenv

Reply:
@benlangmuir Yes sure! I can't see why can't we do that. cc @xedin

Reply:
Sure!